contents/security/Https.md

```diff
@@ -0,0 +1,59 @@
+
+Thus, by using https, we can be sure that even though our network packets are transmitted in an insecure network, sniffers cannot understand the content of our encrypted packets. 
+
+
+Besides providing secure network traffic, https also provides server validation through Certificate Authority(CA) architecture. A detailed explanation on CA is [here](https://www.globalsign.com/en-sg/ssl-information-center/what-are-certification-authorities-trust-hierarchies/) . In short, CA works by issuing the server a digital certificate that can only be produced by CA and client browsers can verify the server with digital certificate the server provides. 
+
+## How to set up https
+In order set up https on your server, you would need to have:
+1. A dedicated IP address for you server.
+2. Purchase a digital certificate from a CA.
+
+For example, one of the CA is [digicert](https://www.digicert.com) and they provide SSL [installation guide](https://www.digicert.com/ssl-certificate-installation.htm) for different platforms. Usually the SSL setup could be found on the CA that you purchased digital signature from.
+
+## Misuse of https
+
+Up to this moment, it seems that nothing could go wrong with https. However, in real life, there could be weakness on the implementation of https.
+
+1. In the implementation of https, the CA does not sign the digital certificate directly, rather it signes the fixed length hashed digest of the digital certificate for efficiency.This introduces the possibility of two different digital certificate with the same hashed digest. Thus, if the attackers manage to forge a fake digital certificate with the same hashed digest of another valid digital certificate, the browser would trust the attackers server and all the servers signed by the attackers. This loop hole occurs more often with SHA-1 hashing algorithm and SHA-1 is no longer used in https after 2016. A more detailed explanation of the problem is (here)[https://www.sott.net/article/275524-Why-HTTPS-and-SSL-are-not-as-secure-as-you-think]. However, the currently trusted SHA-2 might still be vulnerable to this attack.
+
+2. During the hand shake process of https, client and server send each other in plain text the https standard they support and the most secure standard is chosen to be used. However, this can be exploited for https downgrade by a man in the middle attack. The man in the middle could send forged hand shake message to both server and client to indicate the maximum security supported is only 512 bits Diffie Hellman and trick server and client to encrypt with 512 bits Diffie Hellman. At the moment, a 512 bits Diffie Hellman algorithm could be brute forced with sufficient resources. A more detailed description could be found [here](https://weakdh.org/)
```

Comment:
Think cipher suites might be a more accurate term when server and client negotiates the algorithm to be used. Change brute force to computationally feasible, as algorithms to attack DH are more efficient than brute force?